client/extensions/woocommerce/state/sites/settings/products/actions.js

```diff
@@ -79,3 +115,30 @@ export const updateSettingsProducts = ( siteId, settingsData ) => dispatch => {
 			} );
 		} );
 };
+
+/**
+ * Updates settings/products via batch endpoint.
+ *
+ * @param {Number} siteId wpcom site id.
+ * @param {Object}  successAction, success action object
+ * @param {Object}  failureAction, failure action object
+ * @return {Object} Action object
+ */
+export const saveWeightAndDimensionsUnits = ( siteId, successAction, failureAction ) => (
+	dispatch,
+	getState
+) => {
+	const state = getState();
+	if (
+		! areSettingsProductsLoaded( state, siteId ) ||
+		areSettingsProductsLoading( getState(), siteId )
```

Comment:
can you use `state` again here instead of `getState()`?

Reply:
Yes I can. I must be blind :/